### PR TITLE
Ensure Slashing Protection Exports and Keymanager API Work According to Spec

### PR DESCRIPTION
**Pull request by rauljordan**
As defined in the validator [keymanager API standard](https://github.com/ethereum/beacon-APIs/pull/151/files), there are a few intricate details that we must ensure we support and behave as required. This PR adds a few fixes to ensure the behavior is up to spec:

For `ImportKeystores`
- [x] Ensure that every keystore in the `ImportKeystores` endpoint has a corresponding password. Even if using one password to encrypt everything, we should just duplicate that password N times in the request where N = len(keystores)

For `DeleteKeystores`
- [x] If a key is in the DB but not found in the wallet, we should return NOT_ACTIVE
- [x] We should return slashing protection history only for the keys that are in the request

### CODE REVIEW

File: validator/db/kv/eip_blacklisted_keys.go [hunk 1]
```diff
@@ -7,6 +7,42 @@ import (
 	"go.opencensus.io/trace"
 )
 
+// MarkPublicKeysAsDeleted marks the given keys as deleted from the user's wallet, but should still
+// exist in the database to perform EIP-3076 slashing protection exports if requested.
+func (s *Store) MarkPublicKeysAsDeleted(ctx context.Context, publicKeys [][]byte) error {
+	ctx, span := trace.StartSpan(ctx, "Validator.MarkKeysAsDeleted")
+	defer span.End()
+	return s.db.Update(func(tx *bolt.Tx) error {
```
Comment (james-prysm) [inline validator/db/kv/eip_blacklisted_keys.go:15]: if they reimport the key do you remove it from this list?

File: validator/keymanager/imported/delete.go [hunk 2]
```diff
@@ -73,6 +73,8 @@ func (km *Keymanager) DeleteKeystores(
 			deletedKeysStr += fmt.Sprintf(",%#x", bytesutil.Trunc(k))
 		}
 	}
+
+	// Mark the keys as deleted in our database.
```
Comment (rauljordan) [inline validator/keymanager/imported/delete.go:77]:
```suggestion
```

File: validator/rpc/standard_api_test.go [hunk 3]
```diff
@@ -248,39 +242,115 @@ func TestServer_DeleteKeystores(t *testing.T) {
...
 	}
-	publicKeys, err = km.FetchValidatingPublicKeys(ctx)
-	require.NoError(t, err)
-	require.Equal(t, 0, len(publicKeys))
-	require.Equal(t, numAccounts, len(mockJSON.Data))
+	tests := []struct {
+		keys         []*keyCase
+		wantStatuses []ethpbservice.DeletedKeystoreStatus_Status
+	}{
+		{
```
Comment (rauljordan) [inline validator/rpc/standard_api_test.go:265]: Table driven test to match what the specification wants

**Comment by rauljordan** [comment]
Ready @james-prysm

**Review by james-prysm — APPROVED** [review]
will deprecate in favor of keystore standards in the future after implementation is done.

**Pull request merged**